### PR TITLE
added features for company

### PR DESCRIPTION
**Pull request by Oladiposamuel**

### CODE REVIEW

**Comment by Oladiposamuel** [comment]
Company now has all the features  available for candidates too.

**Review by ayodeji — COMMENTED** [review]
Please fix the things i commented on.

**Comment by ayodeji** [comment]
@Oladiposamuel  Please action the comments i put in this pull request, then @yimjohns  and I will review it, then you can merge. Is there any reason you closed it?

**Comment by Oladiposamuel** [comment]
> @Oladiposamuel Please action the comments i put in this pull request, then @yimjohns and I will review it, then you can merge. Is there any reason you closed it?

Sorry, I didn't realise that. Can you review now?

File: public/dashboard/companies/list.php [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+
+    
+    $total_companies = $company->getTotalCompanies();
+    
+    // $candidates_per_page = 10;
+    // // $page = isset($_GET['page']) ? max(1, intval($_GET['page'])) : 1;
+    // $page = isset($_GET['page']) && is_numeric($_GET['page']) ? max(1, (int)$_GET['page']) : 1;
+    // $offset = ($page - 1) * $candidates_per_page;
+
```
[the remark this answers is not in the document]
Comment (ayodeji) [reply public/dashboard/companies/list.php]: @Oladiposamuel  after you fix any issue i commented on - you can add a comment and mark it as resolved

File: classes/Company.php [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            $stmt->bindParam(':contact_person', $this->contact_person);
+
+            if($stmt->execute()) {
+                return true;
+            }
+
+            return false;
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while creating a company in the database! \n", 3, 'error_log');
```
Comment (ayodeji) [inline classes/Company.php:64]: Log the actual exception $e so we can see the issue in the log file for troubleshooting.

File: classes/Company.php [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        try{
+            $query = "SELECT * FROM companies ORDER BY created_at DESC LIMIT :limit OFFSET :offset";
+            $stmt = $this->conn->prepare($query);
+            $stmt->bindParam(':limit', $limit, PDO::PARAM_INT);
+            $stmt->bindParam(':offset', $offset, PDO::PARAM_INT);
+            $stmt->execute();
+            return $stmt;
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while reading a company from the database! \n", 3, 'error_log');
```
Comment (ayodeji) [inline classes/Company.php:78]: Log the actual exception $e so we can see the issue in the log file for troubleshooting.

File: classes/Company.php [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    public function getTotalCompanies(){
+        try{
+            $query = 'SELECT COUNT(*) AS total FROM ' . $this->table;
+            $stmt = $this->conn->prepare($query);
+            $stmt->execute();
+            $total_candidates = $stmt->fetch(PDO::FETCH_ASSOC);
+            return $total_candidates['total'];
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while getting total number of companies from the database! \n", 3, 'error_log');
```
Comment (ayodeji) [inline classes/Company.php:91]: Log the actual exception $e so we can see the issue in the log file for troubleshooting.

File: classes/Company.php [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                specialization LIKE :search";
+            $stmt = $this->conn->prepare($query);
+            $search_param = "%{$search_string}%";
+            $stmt->bindParam(':search', $search_param);
+            $stmt->execute();
+            $row = $stmt->fetch(PDO::FETCH_ASSOC);
+            return $row['total'];
+        } catch (PDOException $e) {
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while getting total number of companies for search from the database! \n", 3, 'error_log');
```
Comment (ayodeji) [inline classes/Company.php:111]: Log the actual exception $e so we can see the issue in the log file for troubleshooting.

File: classes/Company.php [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+            $search_param = "%{$search_string}%";
+            $stmt->bindParam(':search', $search_param);
+            $stmt->bindParam(':limit', $limit, PDO::PARAM_INT);
+            $stmt->bindParam(':offset', $offset, PDO::PARAM_INT);
+            $stmt->execute();
+            return $stmt;
+            
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while searching for company with multiple paramaters in the database! \n", 3, 'error_log');
```
Comment (ayodeji) [inline classes/Company.php:152]: Log the actual exception $e so we can see the issue in the log file for troubleshooting.

**Review by ayodeji — COMMENTED** [review]
Log the actual exception $e so we can see the issue in the log file for troubleshooting. Do this everywhere you used the log file.

File: classes/Company.php (hunk 4 above)
Comment (Oladiposamuel) [reply classes/Company.php:94]: I have done this.

**Review by ayodeji — APPROVED** [review]
Looks good.

**Review by yimjohns — APPROVED** [review]
All changes reviewed and approved. Proceeding with the merge.

**Pull request merged**